skada/deep/base.py

```diff
@@ -621,6 +622,7 @@ def predict_proba(
             The predicted class probabilities.
         """
         X, _ = self._prepare_input(X, sample_domain, sample_weight)
+        self.predict_nonlinearity = _infer_predict_nonlinearity(self)
```

Comment:
I don't like that the instance is modified in the predict function. One poor man solution would be to delete the attribute after the super().predict_proba call

Reply:
I removed the `self.predict_nonlinearity =...`